### ISSUE TITLE
Adding code examples to the documentation

### ISSUE THREAD

**Issue by armandas** [Rahix/avr-hal#590]
I'm thinking to make documentation more useful and accessible by adding code examples. Would you be happy to accept this?
To start with, I would add just a basic example to the documentation of each module, like below:

![image](https://github.com/user-attachments/assets/f936bf28-bddb-493b-b73c-4deca2fc987e)

I would also like to make one PR per crate or possibly per module. What to you think?

**Comment by Rahix** [comment]
Yes please, the documentation is still quite lacking at the moment.  I think it may also be a good idea to link to relevant example programs from the docs.  I think working through this per-crate is fine, it that works for you.
